examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -38,7 +38,7 @@
 public class WordCountIT {
   private static final String DEFAULT_INPUT =
       "gs://apache-beam-samples/shakespeare/winterstale-personae";
-  private static final String DEFAULT_OUTPUT_CHECKSUM = "508517575eba8d8d5a54f7f0080a00951cfe84ca";
+  private static final String DEFAULT_OUTPUT_CHECKSUM = "ebf895e7324e8a3edc72e7bcc96fa2ba7f690def";
```

Comment:
Why did this need to change?

Reply:
I found that after `FileChecksumMatcher` serialized and deserialized, it actually always pass (even change the checksum value). Not sure the root cause, but I guess it may come from [`ShardedFile`](https://github.com/apache/beam/blob/66fcda99c1e5902ff4d0981b887c70f7afe9380a/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/FileChecksumMatcher.java#L114).

After my changes, the `FileChecksumMatcher` start doing verify work and the checksum always "ebf895e7324e8a3edc72e7bcc96fa2ba7f690def" for dataflow runner and spark runner.

Reply:
This is a good example of the danger of using a checksum instead of checking that a few lines of the file match actual content since the checksum is difficult to validate so you can't be sure that the system is working as expected without debugging through it.

Reply:
yes. we can create another matcher like `FileContentMatcher` to verify lines given expected files, and add to ITs like wordcount test and windowedwordcount test. I'd like to create a JIRA and probably fix it in another PR.

Reply:
Create a JIRA to track above suggestion: https://issues.apache.org/jira/browse/BEAM-3810